File: java/src/test/java/problems201_300/UglyNumberTest.java
package problems201_300;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class UglyNumberTest extends SampledTest {
    @Test
    public void test() {
        for (UglyNumber solution : UglyNumber.values()) {
            for (UglyNumberSample sample : getSamples(UglyNumberSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.isUgly(sample.input.n))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: javascript/__tests__/problems201_300/ugly-number.test.js
import {strictEqual} from 'assert';
import SOLUTIONS from '../../src/problems201_300/ugly-number';
import getSamples from '../../../testing/js/src/sample/samples';

describe(
    'UglyNumber',
    () =>
        test(
            'test',
            () =>
                SOLUTIONS.forEach(solution =>
                    getSamples().forEach(sample =>
                        strictEqual(
                            solution.isUgly(sample.input.n),
                            sample.output,
                            sample.message,
                        )
                    ),
                ),
        ),
);
